firmware/library/L1_Drivers/ssp.hpp

```diff
@@ -70,3 +72,215 @@ class SspInterface
     virtual void SetClock(bool polarity, bool phase,
             uint8_t prescaler, uint8_t divider)                     = 0;
 };
+
+class Ssp : public SspInterface
+{
+ public:
+    // SSP register lookup table
+    static LPC_SSP_TypeDef * ssp_registers[2];
+    static LPC_SC_TypeDef * sysclock_register;
+    static constexpr uint8_t kEnableSspPins = 0b010;
+    static constexpr uint8_t kPort0 = 0;
+    static constexpr uint8_t kPin18 = 18;
+    static constexpr uint8_t kPin17 = 17;
+    static constexpr uint8_t kPin15 = 15;
+    static constexpr uint8_t kPin9 = 9;
+    static constexpr uint8_t kPin8 = 8;
+    static constexpr uint8_t kPin7 = 7;
+    static constexpr uint8_t kDataBit = 0;
+    static constexpr uint8_t kFrameBit = 4;
+    static constexpr uint8_t kPolarityBit = 6;
+    static constexpr uint8_t kPhaseBit = 7;
+    static constexpr uint8_t kDividerBit = 8;
+    static constexpr uint8_t kMasterModeBit = 2;
+    static constexpr uint8_t kDataLineIdleBit = 4;
+
+    // Default constructor sets up SSP0 peripheral as SPI master
+    constexpr Ssp() : mosi_(mosi_pin_),
+        miso_(miso_pin_),
+        sck_(sck_pin_),
+
+        mosi_pin_(kPort0, kPin18),   // P0.18
+        miso_pin_(kPort0, kPin17),   // P0.17
+        sck_pin_(kPort0, kPin15),    // P0.15

```

Comment:
I think that the comments can be omitted since its kinda of obvious from the variable names.